x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/utils.ts

```diff
@@ -61,3 +61,50 @@ export const normalizeBoostValues = (boosts: Record<string, RawBoost[]>): Record
       [fieldName]: boostList.map(normalizeBoostValue),
     };
   }, {});
+
+// Our model allows for empty values to be added to boosts. However, the server will not accept
+// empty strings in values. To avoid that, we filter out empty values before sending them to the server.
+
+// I.e., the server will not accept any of the following, so we need to filter them out
+// value: undefined
+// value: []
+// value: ['']
+// value: ['foo', '']
+export const removeEmptyValueBoosts = (
+  boosts: Record<string, Boost[]>
+): Record<string, Boost[]> => {
+  // before:
+  //   { foo: { values: ['a', '', '   '] } }
+  // after:
+  //   { foo: { values: ['a'] } }
+  const trimBoostValues = (fieldBoosts: Boost[]) => {
+    return fieldBoosts.map((boost: Boost) => {
+      if (boost.type !== BoostType.Value) return boost;
+      const valueBoost = boost as ValueBoost;
+      return {
+        ...boost,
+        value: valueBoost.value.filter((v) => v.trim() !== ''),
```

Comment:
I'll probably get made fun of on Twitter for this, but times like these I _almost_ miss working in PHP - `array_filter($some_array)` automatically handled filtering out falsy values for you without any extra steps which was kinda awesome. I don't think it had trimming though, so good call on that part!